submissions/dashakim/memory-game/script.js

```diff
@@ -0,0 +1,90 @@
+const section = document.querySelector("section");
+
+const imagesNames = function(currentImg) {
+  return currentImg
+    .split(".")
+    .slice(0, 1)
+    .join("")
+    .substr(4);
+};
+
+const createFrontFaceCard = function(div) {
+  const frontImage = document.createElement("img");
+  frontImage.classList.add("front-face");
+  frontImage.setAttribute("src", frontImageQuestion);
+  frontImage.setAttribute("alt", "icon-question");
+  div.appendChild(frontImage);
+};
+
+const createBackFaceCard = function(div, el) {
+  const backImage = document.createElement("img");
+  backImage.classList.add("back-face");
+  backImage.setAttribute("src", el);
+  backImage.setAttribute("alt", imagesNames(el));
+  div.appendChild(backImage);
+};
+
+const createCard = function(storage) {
+  storage.map(i => {
+    const divMemoryCard = document.createElement("div");
+    divMemoryCard.classList.add("memory-card");
+    section.appendChild(divMemoryCard);
+    divMemoryCard.dataset.framework = imagesNames(i);
+    createFrontFaceCard(divMemoryCard);
+    createBackFaceCard(divMemoryCard, i);
+  });
+};
+
+createCard(images);
+
+const card = document.querySelectorAll(".memory-card");
+let lockBoard = false;
+let hasFlippedCard = false;
+let firstCard, secondCard;
+
+function flipCard() {
+  if (lockBoard) return;
+  if (this === firstCard) return;
+  this.classList.add("flip");
+  if (!hasFlippedCard) {
+    hasFlippedCard = true;
+    firstCard = this;
+
+    return;
+  }
+
+  secondCard = this;
+  checkForMatch();
+}
+
+function checkForMatch() {
+  let isMatch = firstCard.dataset.framework === secondCard.dataset.framework;
+  isMatch ? disableCards() : unflipCards();
+}
+
+function disableCards() {
+  firstCard.removeEventListener("click", flipCard);
+  secondCard.removeEventListener("click", flipCard);
+  resetBoard();
+}
```

Comment:
Also, since now is no longer a single event listener on each card - there is nothing to remove. The function now just calling another function - so we don't need it anymore.

Reply:
At that time, we removed event listeners from flipped pairs to prevent them from clicking. Now we can do it with adding one more check to `flipCard` function: `if (target.classList.contains('flip')) return` (as one of the possible ways).